src/include/gurt/common.h

```diff
@@ -269,6 +271,8 @@ char *d_realpath(const char *path, char *resolved_path);
 #define D_FREE(ptr)                                                                                \
 	do {                                                                                       \
 		if ((ptr) != NULL) {                                                               \
+			size_t _frs = malloc_usable_size(ptr);                                     \
+			memset(ptr, 0x42, _frs);                                                   \
```

Comment:
Should we put this into a DAOS_BUILD_RELEASE ifndef ?

Reply:
Yeah, I don't think we should land this I just wanted to see if it would work - which it all does apart from one test.